client/js/libs/handlebars/parse.js

```diff
@@ -81,5 +81,8 @@ module.exports = function parse(text) {
 		}
 
 		return fragments;
+	}).join("") + linkParts.map((part) => {
+		const escapedLink = Handlebars.Utils.escapeExpression(part.link);
+		return `<div class="preview" data-url="${escapedLink}"></div>`;
```

Comment:
Yeah, alternatively it was messing with jQuery to insert at the right position depending on the order, etc. so this feels more "natural" but I completely feel you. Nothing we can't update if a better solution presents itself anyway